contracts/protocol/bases/SellerBase.sol

```diff
@@ -20,6 +21,8 @@ contract SellerBase is ProtocolBase, IBosonAccountEvents {
      * Emits a SellerCreated event if successful.
      *
      * Reverts if:
+     * - Caller is not the supplied admin or does not own supplied auth token
+     * - Caller is not the supplied operator and clerk
```

Comment:
Why the check for treasury isn't done:
- frontrunner cannot create seller or DR with fields {admin: victim, operator: victim, clerk: victim, treasury: attacker}. It will fail, since new check will detect that msg.sender (==atacker) does not match the victim's address
- treasury cannot perform any action in the protocol